sql-plugin/src/main/scala/org/apache/spark/sql/rapids/GpuFileFormatDataWriter.scala

```diff
@@ -235,16 +236,22 @@ class GpuSingleDirectoryDataWriter(
     statsTrackers.foreach(_.newFile(currentPath))
   }
 
+  private def doWriteAndClose(scb: SpillableColumnarBatch): Unit = {
```

Comment:
nit: Could we rename this so it is clear that we are updating metrics too?  Having a side effect is important here.